Build failure: protobuf_27

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix-build -A protobuf_27

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310439223

### Relevant log output

```console
➜  nixpkgs git:(master) nix-build -A protobuf_27
this derivation will be built:
  /nix/store/afdj9vzl83irya8jdnfj5s2jg64z9bxv-protobuf-27.5.drv
building '/nix/store/afdj9vzl83irya8jdnfj5s2jg64z9bxv-protobuf-27.5.drv'...
Using versionCheckHook
Running phase: unpackPhase
unpacking source archive /nix/store/qvfv2vgafh79xcxvd2xsgx0igqyi7cbs-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LOCALEDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/share/doc/protobuf -DCMAKE_INSTALL_INFODIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/include -DCMAKE_INSTALL_SBINDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5 -Dprotobuf_USE_EXTERNAL_GTEST=ON -Dprotobuf_ABSL_PROVIDER=package -Dprotobuf_BUILD_SHARED_LIBS=ON
CMake Deprecation Warning at CMakeLists.txt:7 (cmake_policy):
  The OLD behavior for policy CMP0141 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- The C compiler identification is GNU 14.3.0
-- The CXX compiler identification is GNU 14.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
--
-- 27.5.0
-- Performing Test protobuf_HAVE_LD_VERSION_SCRIPT
-- Performing Test protobuf_HAVE_LD_VERSION_SCRIPT - Success
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found ZLIB: /nix/store/z55x0q74zldi64iwamqf8wgrm2iza5rk-zlib-1.3.1/lib/libz.so (found version "1.3.1")
-- Performing Test protobuf_HAVE_BUILTIN_ATOMICS
-- Performing Test protobuf_HAVE_BUILTIN_ATOMICS - Success
CMake Deprecation Warning at third_party/utf8_range/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Configuring done (0.9s)
CMake Error at cmake/libprotobuf-lite.cmake:30 (target_link_libraries):
  Target "libprotobuf-lite" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:303 (include)


CMake Error at cmake/libprotobuf.cmake:33 (target_link_libraries):
  Target "libprotobuf" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:307 (include)


CMake Error at cmake/libprotoc.cmake:20 (target_link_libraries):
  Target "libprotoc" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:312 (include)


CMake Error at cmake/upb_generators.cmake:18 (target_link_libraries):
  Target "protoc-gen-upb" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:322 (include)


CMake Error at cmake/upb_generators.cmake:18 (target_link_libraries):
  Target "protoc-gen-upbdefs" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:322 (include)


CMake Error at cmake/upb_generators.cmake:18 (target_link_libraries):
  Target "protoc-gen-upb_minitable" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:322 (include)


CMake Error at cmake/protoc.cmake:6 (target_link_libraries):
  Target "protoc" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:325 (include)


CMake Error at cmake/tests.cmake:91 (target_link_libraries):
  Target "libtest_common" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:359 (include)


CMake Error at cmake/tests.cmake:108 (target_link_libraries):
  Target "tests" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:359 (include)


CMake Error at cmake/tests.cmake:120 (target_link_libraries):
  Target "fake_plugin" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:359 (include)


CMake Error at cmake/tests.cmake:132 (target_link_libraries):
  Target "test_plugin" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:359 (include)


CMake Error at cmake/tests.cmake:145 (target_link_libraries):
  Target "libtest_common_lite" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:359 (include)


CMake Error at cmake/tests.cmake:156 (target_link_libraries):
  Target "lite-test" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:359 (include)


CMake Error at cmake/tests.cmake:200 (target_link_libraries):
  Target "upb-test" links to:

    absl::if_constexpr

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:359 (include)


-- Generating done (0.2s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


CMake Generate step failed.  Build files cannot be regenerated correctly.
error: Cannot build '/nix/store/afdj9vzl83irya8jdnfj5s2jg64z9bxv-protobuf-27.5.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/h0mdzcxi6cfh0r8p1gdaadszafhzrdp2-protobuf-27.5
       Last 25 log lines:
       >
       > CMake Error at cmake/tests.cmake:200 (target_link_libraries):
       >   Target "upb-test" links to:
       >
       >     absl::if_constexpr
       >
       >   but the target was not found.  Possible reasons include:
       >
       >     * There is a typo in the target name.
       >     * A find_package call is missing for an IMPORTED target.
       >     * An ALIAS target is missing.
       >
       > Call Stack (most recent call first):
       >   CMakeLists.txt:359 (include)
       >
       >
       > -- Generating done (0.2s)
       > CMake Warning:
       >   Manually-specified variables were not used by the project:
       >
       >     CMAKE_EXPORT_NO_PACKAGE_REGISTRY
       >     CMAKE_POLICY_DEFAULT_CMP0025
       >
       >
       > CMake Generate step failed.  Build files cannot be regenerated correctly.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.54, NixOS, 25.11 (Xantusia), 25.11.20251022.01f116e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/sqilaxdxljyalv530z1qrd1q3nd4lyxq-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -8342,7 +8342,10 @@ with pkgs;
         # More recent versions of abseil seem to be missing absl::if_constexpr
         abseil-cpp = abseil-cpp_202407;
       };
-      protobuf_27 = callPackage ../development/libraries/protobuf/27.nix { };
+      protobuf_27 = callPackage ../development/libraries/protobuf/27.nix {
+        # More recent versions of abseil seem to be missing absl::if_constexpr
+        abseil-cpp = abseil-cpp_202407;
+      };
       protobuf_25 = callPackage ../development/libraries/protobuf/25.nix { };
       protobuf_21 = callPackage ../development/libraries/protobuf/21.nix {
         abseil-cpp = abseil-cpp_202103;

